modules/dag-l0/src/main/scala/io/constellationnetwork/dag/l0/http/routes/DelegatedStakesRoutes.scala

```diff
@@ -140,6 +147,144 @@ final case class DelegatedStakesRoutes[F[_]: Async: Hasher](
       .traverse { case DelegatedStakeRecord(delegatedStaking, _, _) => DelegatedStakeReference.of(delegatedStaking) }
       .map(_.getOrElse(DelegatedStakeReference.empty))
 
+  private def toAmount(value: Long): F[Amount] =
+    if (value == 0L) Amount.empty.pure[F]
+    else
+      PosLong
+        .from(value)
+        .pure[F]
+        .map(_.leftMap(new IllegalArgumentException(_)))
+        .flatMap(Async[F].fromEither(_))
+        .map(Amount(_))
+
+  private def getRewardsInfo(
+    lastSnapshot: GlobalIncrementalSnapshot,
+    lastSnapshotInfo: GlobalSnapshotInfo
+  ): F[RewardsInfo] =
+    for {
+      emConfig <- delegatedRewardsConfig.emissionConfig
+        .get(environment)
+        .pure[F]
+        .flatMap(Async[F].fromOption(_, new RuntimeException(s"Could not retrieve emission config for env: $environment")))
+
+      (_, _, totalDelegateStake, currentTotalSupply) <- processDelegations(lastSnapshotInfo)
+      latestDelegateRewardsNoCommission <- getLatestDelegateRewardTotal(lastSnapshot, lastSnapshotInfo)
```

Comment:
Maybe worth a longer discussion at some point on reducing the load on this API by pre-calculating this, but since we're doing this everywhere else probably not worth implementing here.